jquery.facetview.js

```diff
@@ -1581,6 +1582,11 @@ search box - the end user will not know they are happening.
                     $(options.searchbox_class).bindWithDelay('keyup',dosearch,options.freetext_submit_delay);
                 }
 
+                // Properly set size for each facet
+                $.each(options.facets, function(facet){
+                  options.facets[facet]["size"] = options.facet_size;
```

Comment:
That's the problem, it's not being set on the first go round, thats why each facet returns so many elements. The query string building may be broken. https://github.com/msarahan/facetview/blob/weapons/jquery.facetview.js#L1276-L1281

Basically, to get only 10 facets to search, each facet has to have a "size" property the value of which is 10. That isn't set until you set the size by manually clicking on the button. In other words, the items here would have to have an extra key: value, "size": 10
https://github.com/msarahan/facetview/blob/weapons/index.html#L37-L49

Since there is no size variable set, it returns all it can find, which is 100(not set anywhere that I can see).